spec/models/team_spec.rb

```diff
@@ -7,4 +7,9 @@
   it { should validate_presence_of(:owner) }
   it { should have_many(:namespaces) }
 
+  it 'downcases the name on creation' do
+    team = FactoryGirl.create(:team, name: 'TeAm')
```

Comment:
Yes, I agree. I'd do it `before_validation` just for consistency's sake: On the UI we will be performing this validation so the user won't be able to create a team with an invalid name. Therefore, it makes sense to perform such validation also on the server side.

I'm updating this patch to match this reasoning.